src/cubi_tk/irods_common.py

```diff
@@ -106,13 +107,22 @@ def _check_and_gen_irods_files(self, overwrite = False):
         try:
             #check if irodsfile exists
             irodsA_path = self.irods_env_path.parent.joinpath(".irodsA")
+            last_profile_path = self.irods_env_path.parent.joinpath("last_profile.json")
             if (irodsA_path.exists()):
                 self.irodsA_file_found = True
+                if last_profile_path.exists():
+                    with open(last_profile_path) as last_profile_file:
+                        last_used_env = json.load(last_profile_file)["last_used_env"]
+                        overwrite = last_used_env != str(self.irods_env_path) #overwrite irodsA file if last authenticated profile is different to current
+                elif not str(self.irods_env_path).endswith("irods_environment.json"):
```

Comment:
I think this could also be something like `self.irods_env_path.basename != "irods_environment.json"`